logic: add local-prepared logic test config

jordanlewis opened the pull request:
This commit adds a new mode to the logic test harness that runs every
statement as a prepared statement rather than using the default "simple
query" mode of execution.

Release justification: test change

Release note: None

jordanlewis committed:
logic: add local-prepared config by default

This commit adds a new mode to the logic test harness that runs every
statement as a prepared statement rather than using the default "simple
query" mode of execution.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/77068)
<!-- Reviewable:end -->

jordanlewis committed:
logictest: replace scalars with placeholders

This commit enhances the local-prepared logic test configuration to
replace scalars in queries with placeholders, and execute the queries
as prepared statements with the scalars as arguments.

jordanlewis commented:
Hey @ajwerner, I removed the config from the default set. Would this help with your use case?

ajwerner reviewed (commented):
I learned a little later after asking that I didn't really need it. This does seem like a handy tests. I don't know how to interpret the failures (or, largely, lack of failures when expecting them). 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/77068#-:-N2GpAeL3djLljlzjNJM:bjpqjyj)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->
